resource-managers/kubernetes/core/src/test/scala/org/apache/spark/deploy/kubernetes/submit/submitsteps/initcontainer/InitContainerConfigurationStepsOrchestratorSuite.scala

```diff
@@ -44,6 +44,77 @@ class InitContainerConfigurationStepsOrchestratorSuite extends SparkFunSuite {
   private val INIT_CONTAINER_CONFIG_MAP_KEY = "spark-init-config-map-key"
   private val STAGING_SERVER_URI = "http://localhost:8000"
 
+  test ("error thrown if local jars provided without resource staging server") {
+    val sparkConf = new SparkConf(true)
+      .set(KUBERNETES_DRIVER_LABELS, s"$DEPRECATED_CUSTOM_LABEL_KEY=$DEPRECATED_CUSTOM_LABEL_VALUE")
+      .set(s"$KUBERNETES_DRIVER_LABEL_PREFIX$CUSTOM_LABEL_KEY", CUSTOM_LABEL_VALUE)
+
+    assert(sparkConf.get(RESOURCE_STAGING_SERVER_URI).isEmpty)
+
+    intercept[IllegalArgumentException] {
```

Comment:
Can we inspect the correctness of the error message as well? We wouldn't want an `IllegalArgumentException` to come from some other part of the constructor.